### ISSUE TITLE
 Is it possible to use GPU for solving with FP32 quantization in CHOLMOD?

### ISSUE THREAD

**Issue by pcbcos** [DrTimothyAldenDavis/SuiteSparse#798]
**Is your feature request related to a problem? Please describe.**

Currently, CHOLMOD's calculations on GPU can only use FP64, which is double precision. However, in recent years, the FP64 computation speed of NVIDIA consumer-grade graphics cards is 1/64 of their FP32 speed, which would result in slow solving.

**Describe the solution you'd like**

Supporting FP32 in the GPU module of CHOLMOD could potentially significantly improve computational speed, albeit at the cost of sacrificing some precision.

**Comment by DrTimothyAldenDavis** [comment]
Yes, it's possible.  I only recently added FP32 support to the CPU methods in CHOLMOD.  Adding them to the CUDA kernels is something I hope to do in the future.
